custom_model_runner/datarobot_drum/drum/language_predictors/r_predictor/r_predictor.py

```
@@ -104,3 +104,68 @@ def predict(self, input_filename):
+                return None
+            elif _r_is_raw(r_val):
+                return bytes(r_val)
+            elif _r_is_character(r_val):
+                # Any scalar value is returned from R as one element vector,
+                # so get this value.
+                return str(r_val[0])
+            else:
+                raise DrumCommonException(
+                    "Can not convert R value {} type {}".format(r_val, type(r_val))
+                )
+
+        def _rlist_to_dict(rlist):
+            if _r_is_null(rlist):
+                return None
+            return {str(k): _cast_r_to_py(v) for k, v in rlist.items()}
+
+        data_binary_or_text = data
+
+        if UnstructuredDtoKeys.QUERY in kwargs:
+            if isinstance(kwargs[UnstructuredDtoKeys.QUERY], dict):
+                query_kwargs = kwargs[UnstructuredDtoKeys.QUERY]
+                for k, v in query_kwargs.items():
+                    if v is None:
+                        query_kwargs[k] = ro.rinterface.NULL
+                kwargs[UnstructuredDtoKeys.QUERY] = ro.ListVector(query_kwargs)
+
+        # if data_binary_or_text is str it will be auto converted into R character type;
+        # otherwise if it is bytes, manually convert it into byte vector (raw)
+        r_data_binary_or_text = data_binary_or_text
+        if isinstance(data_binary_or_text, bytes):
+            r_data_binary_or_text = ro.vectors.ByteVector(data_binary_or_text)
+
+        kwargs_filtered = {k: v for k, v in kwargs.items() if v is not None}
+        one_or_list = r_handler.predict_unstructured(
+            model=self._model, data=r_data_binary_or_text, **kwargs_filtered
+        )
+        if isinstance(one_or_list, ro.vectors.ListVector):
+            first = _cast_r_to_py(one_or_list[0])
+            second = _rlist_to_dict(one_or_list[1])
+            ret = first, second
+        elif _r_is_null(one_or_list) or _r_is_character(one_or_list) or _r_is_raw(one_or_list):
```

```
else:
   ret = _case_r_to_py(one_or_list), None
```
let it crash
or wrap with try/except and throw a different error message

---

changed this code, so comment not relevant